src/modals/QRCodeModal/QRCodeModal.jsx

```diff
@@ -25,11 +25,13 @@ const QRCodeModal = ({ onClose, profile }) => {
   const [link, setLink] = useState('')
   const [syncStep, setSyncStep] = useState(null)
   const [message, setMessage] = useState(null)
+  const [sp, setSp] = useState(null)
 
   useEffect(() => {
     const startSync = async () => {
       const sp = new SyncProfile()
       await sp.init()
+      setSp(sp)
```

Comment:
why using a state for sp ?

Reply:
In order to get the same instance of SyncProfile instantiated in startSync.